topi/include/topi/transform.h

```diff
@@ -365,6 +365,101 @@ inline Array<Tensor> split(const Tensor& x,
   return result;
 }
 
+/*!
+* \brief strided_slice of a tensor
+*
+* \param x The input tensor
+* \param begin The indices to begin with in the slicing
+* \param end Indicies indicating end of the slice
+* \param stride Specifies the stride values, it can be negative
+* in that case, the input tensor will be reversed in that particular axis
+* \param name The name of the operation
+* \param tag The tag to mark the operation
+*
+* \return A Tensor whose op member is the split operation
+*/
+inline Tensor strided_slice(const Tensor& x,
+                            const Array<Expr> begin,
+                            const Array<Expr> end,
+                            const Array<Expr> stride,
+                            std::string name = "tensor",
+                            std::string tag = kInjective) {
+  size_t src_tensor_dim = static_cast<size_t>(x->shape.size());
+  std::vector<int> begin_val = GetConstIntValues(begin, "begin");
+  std::vector<int> end_val = GetConstIntValues(end, "end");
+  std::vector<int> stride_val = GetConstIntValues(stride, "stride");
+
+  /* User can give begin, end & stride for only few axes, in that case */
+  /* expand the array for begin,end & stride to match num of axes of input data */
+  /* and fill with corresponding default values. */
+  /* If user has not provided stride, then construct stride with default values as 1 */
+  auto expand_axis = [x](std::vector<int> list_ids,
+                         size_t src_tensor_dim, int fill_value, bool end) {
+    if (list_ids.size() < src_tensor_dim) {
+      for (size_t i = list_ids.size(); i < src_tensor_dim; ++i) {
+        if (end) {
+          list_ids.push_back(GetConstInt(x->shape[i]));
+        } else {
+          list_ids.push_back(fill_value);
+        }
+      }
+    }
+    return list_ids;
+  };
+
+  std::vector<int> begin_ids;
+  std::copy(begin_val.begin(), begin_val.end(), std::back_inserter(begin_ids));
+  begin_ids = expand_axis(begin_ids, src_tensor_dim, 0, false);
+
+  std::vector<int> end_ids;
+  std::copy(end_val.begin(), end_val.end(), std::back_inserter(end_ids));
+  end_ids = expand_axis(end_ids, src_tensor_dim, 0, true);
+
+  std::vector<int> stride_ids;
```

Comment:
what does id mean? index? The strides are not indices, maybe just use strides or strides_vec